test/test_sockets.py

```diff
@@ -329,6 +329,14 @@ def test_websocket_send(self):
     with NodeJsWebSocketEchoServerProcess():
       self.btest_exit(test_file('websocket/test_websocket_send.c'), args=['-lwebsocket', '-sNO_EXIT_RUNTIME', '-sWEBSOCKET_DEBUG'])
 
+  # Test Emscripten WebSockets API to send and receive text and binary messages against an echo server.
+  # N.B. running this test requires 'npm install ws' in Emscripten root directory
+  # test in shared memory environment
+  # websocket sending only allowd non shared buffer
+  def test_websocket_send_shared_memory(self):
+    with NodeJsWebSocketEchoServerProcess():
+      self.btest_exit(test_file('websocket/test_websocket_send.c'), args=['-lwebsocket', '-sSHARED_MEMORY', '-sNO_EXIT_RUNTIME', '-sWEBSOCKET_DEBUG'])
```

Comment:
You should be able use `@parameterized` to avoid duplicating the test body.

Something like:

```
  @parameterized({                                                                                                                                                                                                
    '': ([],),                                                                                       
    'shared_memory': (['-sSHARED_MEMORY'],),                             
  }) 
```